### PR TITLE
Fix 1270 port `StringCollectionEditor`

### LINKED ISSUE #1270: Port System.Windows.Forms.Design.StringCollectionEditor
* .NET Core Version: (e.g. 3.0 Preview1, or daily build number, use `dotnet --info`)
* Have you experienced this same bug with .NET Framework?: Yes/No

**Problem description:**
`StringCollectionEditor` from System.Windows.Forms.Design is missing

Refer to https://github.com/dotnet/winforms/issues/1115#issuecomment-504798149

### PR DESCRIPTION
**Pull request by RussKie**
Closes #1270

## Proposed changes

- Port StringCollectionEditor from NET Fx
- Fix xml-docs
- Fix names
- Remove unused usings
- Remove `this.`
- Add missing braces
- Use arrow functions, where appropriate

### After
image

## Test methodology 

- TODO

### CODE REVIEW

**Comment by RussKie** [comment]
@M-Lipin @vladimir-krestov I'll need your help with this one

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#define NEEDHELPBUTTON
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs]: Is this macro used anywhere?

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                    }
+                }
+
+                if (!dirty)
+                {
+                    DialogResult = DialogResult.Cancel;
+                    return;
+                }
+
+                // ASURT #57372
```
Comment (danmosemsft) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs]: Remove bug numbers ... ASURT is ~2002 era ("application server universal runtime" I think)

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs]: I couldn't find it, removed

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs (hunk 2 above)
Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/StringCollectionEditor.cs]: Done

**Comment by RussKie** [comment]
@M-Lipin once your changes to `StringCollectionEditor` are merged into NET Fx, please port them here as well. Thank you

**Comment by M-Lipin** [comment]
> @M-Lipin once your changes to StringCollectionEditor are merged into NET Fx, please port them here as well. Thank you

Will do, thanks.

**Pull request merged**